src/contract.rs

```diff
@@ -736,18 +619,34 @@ pub fn handle_jackpot<S: Storage, A: Api, Q: Querier>(
         return Err(StdError::generic_err("Empty contract balance"));
     }
     // Ensure the contract have sufficient balance to handle the transaction
-    if balance.amount < jackpot_amount {
+    if balance.amount < state.jackpot_reward {
         return Err(StdError::generic_err("Not enough funds in the contract"));
     }
-    // Ensure there is some reward to send
-    if jackpot_amount.is_zero() {
-        return Err(StdError::generic_err("No jackpot to claim, try next time"));
-    }
+
+    // TODO: maybe save winner count somewhere?
+    let winners = all_winners(&deps.storage, state.lottery_counter)?.len();
+
+    let total_prize = rewards.clone().ranks.into_iter().fold(0u128, |acc, rank| {
+        let prize = state
+            .jackpot_reward
+            .mul(Decimal::percent(
+                // TODO claim.rank checked anywhere? this can cause panic
+                state.prize_rank_winner_percentage[rank as usize - 1] as u64,
+            ))
+            .u128()
+            / winners as u128;
+        acc + prize
+    });
+
+    // update the winner to claimed true
+    rewards.claimed = true;
+    winner_storage(&mut deps.storage, state.lottery_counter)
```

Comment:
same as before in L599, checking the wrong lottery round will result in error i propose to fixe with the previous fix of line L599 with the variable last_lottery_counter_round
```suggestion
    winner_storage(&mut deps.storage, last_lottery_counter_round)
```